### PR TITLE
feat!: upgrade boto code to boto3.

### PR DESCRIPTION
**Pull request by awais786**
**part4:**

Fixing this `tubular/admin/boto_to_add_ingress.py` and adding its tests.

This PR need review from domain expert.

### CODE REVIEW

File: tubular/admin/boto_to_add_ingress.py [hunk 1]
```diff
@@ -45,90 +45,105 @@ def add_ingress_rule(dry_run, go_agent_security_group, go_agent_security_group_o
...
             ))
             raise
 
     # Construct a fake security group for the prod-tools-goagent-sg security group in the edx-tools account.
     # This group will be used to grant the go-agents ingress into the ASG's VPCs.
-    go_agent_security_group = boto.ec2.securitygroup.SecurityGroup(
-        name=go_agent_security_group_name,
-        owner_id=go_agent_security_group_owner,
-        id=go_agent_security_group
+    go_agent_security_group = ec2.create_security_group(
```
Comment (awais786) [inline tubular/admin/boto_to_add_ingress.py:87]: Need some expert review here.

File: tubular/admin/boto_to_add_ingress.py [hunk 2]
```diff
@@ -45,90 +45,105 @@ def add_ingress_rule(dry_run, go_agent_security_group, go_agent_security_group_o
...
             yes_no = six.moves.input("Apply the change to this security group? [Yes]")
             if yes_no in ("", "y", "Y", "yes"):
-                sec_group.authorize(
-                    ip_protocol='tcp',
-                    from_port=22,
-                    to_port=22,
-                    src_group=go_agent_security_group,
-                    dry_run=dry_run
+                # ssh access
+                ec2.authorize_security_group_ingress(
```
Comment (awais786) [inline tubular/admin/boto_to_add_ingress.py:121]: Need expert review here.

File: tubular/admin/boto_to_add_ingress.py
Comment (matthugs) [inline tubular/admin/boto_to_add_ingress.py:1]: I can't find uses of this file. I think we may be able to delete it rather than converting it.

**Review by matthugs — CHANGES_REQUESTED** [review]

**Comment by awais786** [comment]
@dianakhuang your opinion about this file. If it has no usage i can delete it.

**Pull request closed without merging**